Mail/Views/Thread/ThreadView.swift

```diff
@@ -41,14 +41,23 @@ struct ThreadView: View {
 
     @State private var displayNavigationTitle = false
     @State private var replyOrReplyAllMessage: Message?
+    @State private var messagesToMove: [Message]?
+    @State private var nearestFlushAlert: FlushAlertState?
+    @State private var toolbarActions: [Action]
 
     @State private var isShowingExternalTagAlert = false
 
     @ObservedRealmObject var thread: Thread
 
-    @State private var nearestFlushAlert: FlushAlertState?
-
-    private let toolbarActions: [Action] = [.reply, .forward, .archive, .delete]
+    init(thread: Thread) {
+        self.thread = thread
+        _toolbarActions = thread.folder?.role == .archive ? State(wrappedValue: [
```

Comment:
Maybe you can create two static properties ? standardActions and archiveActions?